ci/jenkinsfiles/release.groovy

```diff
@@ -0,0 +1,335 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Contributors:
+*     Abdoul BA <aba@nuxeo.com>
+*/
+
+def appName = 'nuxeo-coldstorage'
+def pipelineLib
+def repositoryUrl = 'https://github.com/nuxeo/nuxeo-coldstorage/'
+
+properties([
+  [
+    $class: 'BuildDiscarderProperty',
+    strategy: [
+      $class: 'LogRotator',
+      daysToKeepStr: '15', numToKeepStr: '10',
+      artifactNumToKeepStr: '5'
+    ]
+  ],
+  [
+    $class: 'GithubProjectProperty', projectUrlStr: repositoryUrl
+  ],
+  disableConcurrentBuilds(),
+])
+
+String currentVersion() {
+  return readMavenPom().getVersion()
+}
+
+String getReleaseVersion(String version) {
+  return version.replace('-SNAPSHOT', '')
+}
+
+pipeline {
+  agent {
+    label 'builder-maven-nuxeo-11'
+  }
+  parameters {
+    string(name: 'rcVersion', description: 'Version to be promoted')
+    string(name: 'reference', description: 'Reference branch to be bumped after releasing')
+    booleanParam(
+      name: 'dryRun', defaultValue: true,
+      description: 'if true all steps will be run without publishing the artifact'
+    )
+  }
+  environment {
+    APP_NAME = "${appName}"
+    BACKEND_FOLDER = "${WORKSPACE}/nuxeo-coldstorage"
+    BRANCH_NAME = GIT_BRANCH.replace('origin/', '')
+    BRANCH_LC = "${BRANCH_NAME.toLowerCase().replace('.', '-')}"
+    CONNECT_PROD_URL = 'https://connect.nuxeo.com/nuxeo'
+    CHART_DIR = 'ci/helm/preview'
+    CURRENT_VERSION = currentVersion()
+    ENABLE_GITHUB_STATUS = 'false'
+    FRONTEND_FOLDER = "${WORKSPACE}/nuxeo-coldstorage-web"
+    JENKINS_HOME = '/root'
+    MAVEN_DEBUG = '-e'
+    MAVEN_OPTS = "${MAVEN_OPTS} -Xms512m -Xmx3072m"
+    ORG = 'nuxeo'
+    VERSION = getReleaseVersion(CURRENT_VERSION)
+  }
+  stages {
+    stage('Check parameters') {
+      steps {
+        script {
+          if (!params.rcVersion || params.rcVersion == '') {
+            currentBuild.result = 'ABORTED'
+            currentBuild.description = 'Missing required version parameter, aborting the build.'
+            error(currentBuild.description)
+          }
+
+          if (!params.reference || params.reference == '') {
+            currentBuild.result = 'ABORTED'
+            currentBuild.description = 'Missing required reference parameter, aborting the build.'
+            error(currentBuild.description)
+          } else {
+            echo '''
+              ----------------------------------------
+              Update Reference Branch
+              ----------------------------------------
+            '''
+            env.REFERENCE_BRANCH = params.reference
+          }
+
+          if (params.dryRun && params.dryRun != '') {
+            env.DRY_RUN_RELEASE = params.dryRun
+            if (env.DRY_RUN_RELEASE == 'true') {
+              env.SLACK_CHANNEL = 'infra-napps'
+            }
+          } else {
+            env.SLACK_CHANNEL = 'pr-napps'
+            env.DRY_RUN_RELEASE = 'false'
+          }
```

Comment:
Perhaps we discussed this point int the past, but can we improve a little bit this part. If I am understanding correctly `params.dryRun` can `true` otherwise it will be considered as false no matter its value, isn't it?

Reply:
indeed.
params.dryRun is a Jenkins boolean. it can be true or false.
Unfortunately false mean it isn't not defined.

Reply:
@doulba I think we can keep as it's for now. Can please make a notes about this kind of things. We will try to implement them when we will start our sharing library. You can do it on the ticket that Nuno created at time :)